Code/GraphMol/RascalMCES/PartitionSet.h

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (C) David Cosgrove 2023
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#ifndef RASCALMCES_PARTITION_SET_H
+#define RASCALMCES_PARTITION_SET_H
+
+#include <map>
+#include <vector>
+
+namespace RDKit {
+
+namespace RascalMCES {
+
+class PartitionSet {
+ public:
+  // Make a partition set from the modular product and the labels
+  // of the vertices from the first graph.  Each element in vtxPairs
+  // has a row/column in modProd.  The partitions are sorted
+  // into descending order of sizes. Because that uses std::sort,
+  // the final order may differ from compiler to compiler, because
+  // the standard doesn't define what should happen in the case of
+  // duplicate values.  It is quite common to have partitions of the
+  // same size.
+  PartitionSet(const std::vector<std::vector<char>> &modProd,
+               const std::vector<std::pair<int, int>> &vtxPairs,
+               const std::vector<unsigned int> &vtx1Labels,
+               const std::vector<unsigned int> &vtx2Labels,
+               unsigned int lowerBound);
+
+  bool isEmpty() const { return d_parts.empty(); }
+
+  size_t numParts() const { return d_parts.size(); }
+
+  // Compute the upper bound on the clique that can be extracted from
+  // the current partition.
+  int upperBound();
+
+  // Print the partitions to os.  Very useful for debugging, but not
+  // used otherwise.
+  void printPartitions(std::ostream &os) const;
```

Comment:
why not just define `std::ostream& operator<<(std::ostream& target, const PartitionSet & pt)`